### PR TITLE
Add System.Composition.Runtime tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

**Comment by hughbe** [comment]
TestLinux x64 Release Build

**Comment by hughbe** [comment]
TestLinux x64 Release Build

**Comment by hughbe** [comment]
Test Linux x64 Release Build

File: src/System.Composition.Runtime/tests/CompositionContextTests.cs [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            });
+            if (success)
+            {
+                Assert.Equal(10, context.GetExport<int>());
+            }
+            else
+            {
+                Assert.Throws<CompositionFailedException>(() => context.GetExport<int>());
+            }
+        }
```
Comment (stephentoub) [inline src/System.Composition.Runtime/tests/CompositionContextTests.cs]: It feels like a lot of these tests share the same structure and just have minor differences between them.  Is it possible to condense these via a helper that contains most of the boilerplate, or are they different enough that doesn't work?

Comment (hughbe) [inline src/System.Composition.Runtime/tests/CompositionContextTests.cs]: The difference in test structure is the overload they call. However, they all pipe through to the same end function, so we can test them all together

**Pull request merged**